From: Lee, Chun-Yi <jlee@suse.com>
Date: Mon Nov 30 15:23:26 2015 +0100
Subject: efifb: Fix KABI of screen_info struct

Patch-mainline: never, KABI fix
References: bsc#973499

Backported ae2ee627 patch from v4.4-rc1 kernel to enable EFI framebuffer
for text mode installation because some EFI machines responses to GOP
protocol a 64-bit address.

The backported patch broke KABI because it adds "__u32 ext_lfb_base"
element to screen_info structure. This patch resolves this issue.

Signed-off-by: Lee, Chun-Yi <jlee@suse.com>
---
 include/uapi/linux/screen_info.h |    4 ++++
 1 file changed, 4 insertions(+)
--- a/include/uapi/linux/screen_info.h
+++ b/include/uapi/linux/screen_info.h
@@ -43,8 +43,12 @@ struct screen_info {
 	__u16 pages;		/* 0x32 */
 	__u16 vesa_attributes;	/* 0x34 */
 	__u32 capabilities;     /* 0x36 */
+#ifdef __GENKSYMS__
+	__u8  _reserved[6];	/* 0x3a */
+#else
 	__u32 ext_lfb_base;     /* 0x3a */
 	__u8  _reserved[2];     /* 0x3e */
+#endif
 } __attribute__((packed));
 
 #define VIDEO_TYPE_MDA		0x10	/* Monochrome Text Display	*/
